src/main/scala/org/neo4j/spark/service/SchemaService.scala

```diff
@@ -158,6 +163,104 @@ class SchemaService(private val options: Neo4jOptions, private val jobId: String
+     * for the sequence of counts returned
+     */
+    val query = "CALL apoc.meta.stats() yield labels RETURN labels"
+    val map = session.run(query).single()
+      .asMap()
+      .asScala
+      .get("labels")
+      .getOrElse(Collections.emptyMap())
+      .asInstanceOf[util.Map[String, Long]].asScala
+    map.filterKeys(k => options.nodeMetadata.labels.contains(k))
+      .values.min
+  } catch {
+    case e: ClientException =>
+      e.code match {
+        case "Neo.ClientError.Procedure.ProcedureNotFound" => {
+          // TODO get back to Cypher DSL
+          options.nodeMetadata.labels
+            .map(label => {
+              val query = s"""MATCH (${Neo4jUtil.NODE_ALIAS}:$label)
+                             |RETURN count(${Neo4jUtil.NODE_ALIAS}) AS count
+                             |""".stripMargin
+              val count = session.run(query).single().get("count")
+              if (count.isNull) 0L else count.asLong()
+            })
+            .min
+        }
+      }
+  }
+
+  def countForRelationship(): Long = try {
+    val query = "CALL apoc.meta.stats() yield relTypes RETURN relTypes"
+    val map = session.run(query).single()
+      .asMap()
+      .asScala
+      .get("relTypes")
+      .getOrElse(Collections.emptyMap())
+      .asInstanceOf[util.Map[String, Long]]
+      .asScala
+    val minFromSource = options.relationshipMetadata.source.labels
+      .map(label => map.get(s"(:$label)-[:${options.relationshipMetadata.relationshipType}]->()").getOrElse(Long.MaxValue))
+      .min
+    val minFromTarget = options.relationshipMetadata.target.labels
+      .map(label => map.get(s"()-[:${options.relationshipMetadata.relationshipType}]->(:$label)").getOrElse(Long.MaxValue))
+      .min
+    Math.min(minFromSource, minFromTarget)
+  } catch {
+    case e: ClientException =>
+      e.code match {
+        case "Neo.ClientError.Procedure.ProcedureNotFound" => {
+          // TODO get back to Cypher DSL when rand function will be available
+          val minFromSource = options.relationshipMetadata.source.labels
+            .map(label => session.run(
+              s"""MATCH (:$label)-[${Neo4jUtil.RELATIONSHIP_ALIAS}:${options.relationshipMetadata.relationshipType}]->()
+                 |RETURN count(${Neo4jUtil.RELATIONSHIP_ALIAS}) AS count
+                 |""".stripMargin).single().get("count").asLong())
+            .min
+          val minFromTarget = options.relationshipMetadata.target.labels
+            .map(label => session.run(s"""MATCH ()-[${Neo4jUtil.RELATIONSHIP_ALIAS}:${options.relationshipMetadata.relationshipType}]->(:$label)
+                                         |RETURN count(${Neo4jUtil.RELATIONSHIP_ALIAS}) AS count
+                                         |""".stripMargin).single().get("count"))
+            .map(count => if (count.isNull) 0L else count.asLong())
+            .min
+          Math.min(minFromSource, minFromTarget)
```

Comment:
why min and not max?

Reply:
because I use the most selective, if we have this use use-case:

```scala
val df = ss.read.format(classOf[DataSource].getName)
      .option("url", SparkConnectorScalaSuiteIT.server.getBoltUrl)
      .option("relationship", "BOUGHT")
      .option("relationship.nodes.map", "false")
      .option("relationship.source.labels", ":Person:RegisteredCustomer")
      .option("relationship.target.labels", ":Product")
      .load()
```

This will get three counts:

- `(:Person)-[:BOUGHT]->()` => 1000 rows
- `(:RegisteredCustomer)-[:BOUGHT]->()` => 250 rows
- `()-[:BOUGHT]->(:Product)` => 1000

we use the most selective.